guide/src/exception.md

```diff
@@ -123,7 +121,7 @@ To check the type of an exception, you can similarly do:
 # use pyo3::prelude::*;
 # Python::with_gil(|py| {
 # let err = PyTypeError::new_err(());
-err.is_instance::<PyTypeError>(py);
```

Comment:
Ah, just been reminded from this that we probably need to make equivalent changes to `PyErr`?

Either that or we could deprecate `Err::is_instance()` too (and advise just using `err.pvalue(py).is_instance_of::<PyTypeError>()`).